Scripts/extras/pull_and_build_from_git.py

```diff
@@ -564,12 +564,18 @@ def build_and_install_cmake(self):
         if cmake_install_filter:
             # If there is a custom install filter, then we need to install to another temp folder and copy over based on the filter rules
             install_target_folder = self.base_temp_folder / 'working_install'
-            if not install_target_folder.is_dir():
-                install_target_folder.mkdir(parents=True)
         else:
             # Otherwise install directly to the target
             install_target_folder = self.build_install_folder
 
+        install_target_folder = install_target_folder.resolve().absolute()
+
+        if self.clean_build and install_target_folder.is_dir():
+            delete_folder(install_target_folder.absolute().resolve())
+
+        if not install_target_folder.is_dir():
+            install_target_folder.mkdir(parents=True)
```

Comment:
mdkir has an argument `exist_ok` which you can set to true that would remove the need for the check. However, there should be a check for `install_target_folder.is_file()` instead

```suggestion
        if install_target_folder.is_file():
            raise BuildError(f'Error: Target folder {install_target_folder} is a file')
        install_target_folder.mkdir(parents=True, exist_ok=True)
```